scanner.go

```diff
@@ -98,12 +98,14 @@ func dbRead(path string, maxRetry int, output string) map[string]*share.ScanVuln
 	}
 }
 
-func connectController(path, advIP, joinIP, selfID string, advPort uint32, joinPort uint16) {
+func connectController(path, advIP, joinIP, selfID string, advPort uint32, joinPort uint16, period, retryMax int, doneCh chan bool) {
```

Comment:
optional: I know we have many `joinIP` everywhere, but it can actually contains service name of controller. Maybe consider renaming it to `joinHost`?

Reply:
good idea.

Reply:
It is not a Host target. The joinServiceAddr or joinServiceIP is better. It is okay to keep joinIP, too.

Reply:
I like `joinServiceAddr`.  Considering scenarios like docker, how about `joinAddr`?